doc/source/whatsnew/v0.18.0.txt

```diff
@@ -318,6 +319,228 @@ other anchored offsets like ``MonthBegin`` and ``YearBegin``.
+
+.. ipython:: python
+
+   r[['A','B']].agg(['mean','sum'])
+
+Upsampling
+''''''''''
+
+.. currentmodule:: pandas.tseries.resample
+
+Upsampling operations take you from a higher frequency to a lower frequency. These are now
+performed with the ``Resampler`` objects with :meth:`~Resampler.backfill`,
+:meth:`~Resampler.ffill`, :meth:`~Resampler.fillna` and :meth:`~Resampler.asfreq` methods.
+
+.. ipython:: python
+
+   s = Series(np.arange(5,dtype='int64'),
+              index=date_range('2010-01-01', periods=5, freq='Q'))
+   s
+
+Previously
+
+.. code-block:: python
+
+   In [6]: s.resample('M', fill_method='ffill')
+   Out[6]:
+   2010-03-31    0
+   2010-04-30    0
+   2010-05-31    0
+   2010-06-30    1
+   2010-07-31    1
+   2010-08-31    1
+   2010-09-30    2
+   2010-10-31    2
+   2010-11-30    2
+   2010-12-31    3
+   2011-01-31    3
+   2011-02-28    3
+   2011-03-31    4
+   Freq: M, dtype: int64
+
+New API
+
+.. ipython:: python
+
+   s.resample('M').ffill()
+
+.. note::
+
+   In the new API, you can either downsample OR upsample. The prior implementation would allow you to pass an aggregator function (like ``mean``) even though you were upsampling, providing a bit of confusion.
+
+
+.. _whatsnew_0180.breaking.aggregation:
+
+Aggregation API clarifictions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+As part of the new API for :ref:`window functions <whatsnew_0180.enhancements.moments>` and :ref:`resampling <whatsnew_0180.resample>`, aggregation functions have been
+clarified. (:issue:`9052`). A full set of examples are presented in :ref:`groupby <groupby.aggregation>`.
+
+Here are the available methodologies and changes for these aggregations.
+
+.. ipython:: python
+
+   df = pd.DataFrame({'A': ['foo', 'bar', 'foo', 'bar',
+                            'foo', 'bar', 'foo', 'foo'],
+                      'B': ['one', 'one', 'two', 'three',
+                            'two', 'two', 'one', 'three'],
+                      'C': np.random.randn(8),
+                      'D': np.arange(8)})
+
+   grouped = df.groupby(['A', 'B'])
+
+Passing a single function or a list of aggregation functions
+
+.. ipython:: python
+
+   grouped[['D','C']].agg(['sum','mean'])
+
+
+Passing a single level dictionary will perform aggregations and name
+the result columns as the keys.
+
+.. ipython:: python
+
+   grouped['D'].agg({'result1': np.sum,
+                     'result2': np.mean})
+
+You can also pass a single level dictionary where the keys are the column names.
+
+.. ipython:: python
+
+   grouped.agg({'C': 'sum', 'D': 'std'})
+
+Finally you can pass a nested dictionary to enable specific functions on specific
+columns and have renaming.
+
+.. ipython:: python
+
+   grouped.agg({'C': {'summer': 'sum'}, 'D' : {'stddev':'std'}})
+
+The change in 0.18.0 is to disallow the following, which could be ambiguous,
+and will now raise a ``SpecificationError``.
+
+.. code-block:: python
+
+   grouped.agg({'summer': {'C': 'sum'}, 'stddev' : {'D':'std'}})
```

Comment:
I don't think this is correct. I think you only allowed this in the rolling refactor? It that case it has not yet been in a released version, so it is not a change. 
(this example raises with KeyError in 0.17.1)